contracts/optimize/OptimizedStorageManager.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./SlotHelper.sol";
+import "../StorageHelper.sol";
+import "../StorageSlotSelfDestructable.sol";
+
+contract OptimizedStorageManager {
+    uint8 internal constant NO_EXIST = 0;
+    uint8 internal constant IN_SLOT = 1;
+    uint8 internal constant IN_CONTRACT_CODE = 2;
+    uint8 internal constant WHERE_STORE_ERROR = 255;
+
+    uint256 internal constant SLOT_LIMIT = 220;
```

Comment:
Looks like we could eliminate StorageManager by setting SLOT_LIMIT as immutable (and set it in ctor).  When SLOT_LIMIT == 0, then we have StorageManager without slots.

Or just copy the code to StorageManager with additional code to check SLOT_LIMIT.